Merchant unilateral close

marsella opened the pull request:
This adds an outline of merchant's unilateral close, including the CLI to initiate it and indications of how the db and escrow agent should be called.

It is dependent on #110 and #124, and as such will probably get rebased a bunch until those branches are stable and merged. 
It includes the merchant close functionality from #113. **Update**: now stable.

A couple notes:
- what's the status of specifying channel ids by prefixes? Currently the command line argument expects a ChannelId; what will happen if it only gets a prefix? Do the sqlite encode/decode functions in libzkchannels take care of this? @yaymukund 
- related: I have not tested the CLI because I haven't been able to establish a channel locally. I think @kwf knows about this situation; it might be worthwhile to delegate fixing this to someone and/or share the setup required.
- We have discussed "outputting" the final balances of the channel, but I don't think there's currently a space to store them in the database. I've noted places where we would need to save them. See boltlabs-inc/zkchannels-spec#51

marsella committed:
add infra to call merchant close #120

marsella committed:
add merchant functionality and improved docs #123

marsella committed:
add merch logic for merch close #120

- fixes a compile error from outdated stuff
- documents a bunch of merchant calls to blockchain
- also makes names more consistent (finalize_X for any completed close operation)

marsella committed:
add customer functionality for merch close #120

indomitableSwan reviewed (changes requested):
Minor comments on customer.

Some issues and questions regarding merchant.

marsella committed:
improve docs for merch-close #120

indomitableSwan reviewed (changes requested):
The documentation would be improved if additional commentary that clarifies when events are expected to occur, i.e. during which type(s) of close flow.

marsella committed:
clarify dispute status, confirmation depth #120

clarifies the difference between confirmation on chain (e.g. it appears
in a block) and at an appropriate depth; uses them correctly.

adds dispute status to customer and flows for initiating it.

marsella committed:
note unused function #120

marsella committed:
clarify documentation #120

These changes are by @indomitableSwan
